airflow/providers/openlineage/conf.py

```diff
@@ -65,6 +65,19 @@ def custom_extractors() -> set[str]:
     return set(extractor.strip() for extractor in option.split(";") if extractor.strip())
 
 
+@cache
+def custom_facet_functions() -> set[str]:
+    """[openlineage] custom_facet_functions."""
+    option = conf.get(_CONFIG_SECTION, "custom_facet_functions", fallback="")
+    if not option:
+        option = os.getenv("OPENLINEAGE_CUSTOM_FACET_FUNCTIONS", "")
```

Comment:
I think good question is whether we can do it without global configs? I think this might be good to be configurable on task/dag/global level, rather than fixed for whole Airflow instance. It goes well with our effort to have lineage be incrementally enabled.

Reply:
>I don't think the OPENLINEAGE_CUSTOM_FACET_FUNCTIONS variable is necessary, we can stick to Airflow config. We use OPENLINEAGE_ variables in some other configs as they are legacy solutions that still need to be supported. For new features we can omit them.

Addressed in ec330478bbefc00118effe6ec9f4930907e3d273

Reply:
>I think good question is whether we can do it without global configs? I think this might be good to be configurable on task/dag/global level, rather than fixed for whole Airflow instance. It goes well with our effort to have lineage be incrementally enabled.

User can choose to `return None` in the custom facet function, if they don't want to add the custom facet for certain tasks/dags. Thinking if the complexity is going to increase if we handle the task/dag/global level settings at config level.